sopel/irc/__init__.py

```diff
@@ -584,15 +592,28 @@ def say(self, text, recipient, max_messages=1):
             # based on how long it's been since our last message to recipient
             if not recipient_stack['flood_left']:
                 recipient_stack['flood_left'] = min(
-                    self.config.core.flood_burst_lines,
-                    int(elapsed) * self.config.core.flood_refill_rate)
+                    flood_burst_lines,
+                    int(elapsed) * flood_refill_rate)
 
             # If it's too soon to send another message, wait
             if not recipient_stack['flood_left']:
-                penalty = float(max(0, len(text) - 50)) / 70
-                wait = min(self.config.core.flood_empty_wait + penalty, 2)  # Maximum wait time is 2 sec
+                penalty = 0
+
+                if flood_penalty_ratio > 0:
```

Comment:
We _could_ change this to `!= 0` in conjunction with the documentation changes I suggested regarding having _only_ 0 represent the "off" state, but it's definitely safer against unintended behavior if negative numbers are ignored.